### DISCUSSION TITLE
New c# 11 required keyword behavior

### DISCUSSION THREAD

**Discussion by ebfortin** [dotnet/csharplang#6237] [General] [answered]
I just installed .NET 7 preview 5 and am playign with the required keyword. In one instance I have a parameterless constructor that set a required property to some value. However I get a CS9035 error (Required member must be set in the obejct initializer or attribute constructor). I wasn't expecting that. I thought setting it in the constructor would make the required requriement fulfilled.

Is this the intended behavior?

**Comment by HaloFour** [comment, marked as answer]
I believe you need to apply the `SetsRequiredMembers` attribute to the constructor: https://github.com/dotnet/csharplang/blob/main/proposals/required-members.md#setsrequiredmembersattribute

**Reply by ebfortin** [reply]
Cool there is a solution to my use case even if what I describe is the intended behavior. Thanks.

**Reply by DavidThielen** [reply]
that link is bad

**Reply by 333fred** [reply]
https://github.com/dotnet/csharplang/blob/main/proposals/csharp-11.0/required-members.md is the moved link.

**Comment by 333fred** [comment]
> Is this the intended behavior?

Yes, that is correct. Required members are a part of the public contract of your constructor, and we do not do any inference of public contracts in C#.
